Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -38,18 +45,45 @@ namespace WebCore {
 
 WTF_MAKE_ISO_ALLOCATED_IMPL(CookieStore);
 
-Ref<CookieStore> CookieStore::create()
+Ref<CookieStore> CookieStore::create(ScriptExecutionContext& context)
```

Comment:
This should take a `Document&`, otherwise your downcast<Document> below looks unsafe.